great_expectations/compatibility/numpy.py

```diff
@@ -0,0 +1,33 @@
+from __future__ import annotations
+
+from typing import Optional, Union
+
+import numpy as np
+from numpy import typing as npt
+from packaging import version
+
+
+def numpy_quantile(
+    a: npt.NDArray, q: float, method: str, axis: Optional[int] = None
+) -> Union[np.float64, npt.NDArray]:
```

Comment:
Nit - `int | None` and `np.float64 | npt.NDArray` (but `ruff` may take care of this?)

Reply:
Good catch, fixed in https://github.com/great-expectations/great_expectations/pull/7647. Ruff does not complain, maybe we need to add UP007? https://beta.ruff.rs/docs/rules/#pyupgrade-up

Reply:
Ah looks like we disable that rule: https://github.com/great-expectations/great_expectations/blob/develop/pyproject.toml#L248. We can probably polish and ratchet or mass fix these.